app/scripts/lib/multichain-api/multichainMethodCallValidator.ts

```diff
@@ -0,0 +1,70 @@
+import { MultiChainOpenRPCDocument } from '@metamask/api-specs';
+import { rpcErrors } from '@metamask/rpc-errors';
+import { ContentDescriptorObject, MethodObject, OpenrpcDocument } from '@open-rpc/meta-schema';
+import dereferenceDocument from '@open-rpc/schema-utils-js/build/dereference-document';
+import { makeCustomResolver } from '@open-rpc/schema-utils-js/build/parse-open-rpc-document';
+import {ValidationError, Validator} from 'jsonschema'
+
+
+const transformError = (error: ValidationError, param: ContentDescriptorObject, got: any) => {
+  // if there is a path, add it to the message
+  const message = param.name
+    + (error.path.length > 0 ? "." + error.path.join('.') : "")
+    + " " + error.message;
+
+  return {
+    code: -32602, // TODO: could be a different error code or not wrapped in json-rpc error, since this will also be wrapped in a -32602 invalid params error
+    message,
+    data: {
+      param: param.name,
+      path: error.path,
+      schema: error.schema,
+      got,
+    }
+  }
+}
+
+var v = new Validator();
+
+const dereffedPromise = dereferenceDocument(MultiChainOpenRPCDocument as any, makeCustomResolver({}));
+export const multichainMethodCallValidator = async (method: string, params: any) => {
+  const dereffed = await dereffedPromise;
```

Comment:
```suggestion
  await dereffedPromise;
```

Reply:
i need the result of `dereffedPromise`.

im assigning it to `dereffedPromise` to deref only once and its async so this gets the result of the promise each time.